### PR TITLE
Adding Refs Packet: Elementwise Unary Ops V

### PR DESCRIPTION
**Pull request by kevinstephano**
This includes the following refs:

* hardsigmoid
* logsigmoid
* rrelu
* silu
* softsign

There are a couple things that need to be checked.
1. `softsign` had some weirdness in that the OpInfo tests were checking integer ops and type promotion was working oddly as `true_divide` promotes float to int. Generically, promoting `softsign` from int to float causes differences in the math because the point where the promotion happens can impact the math.  For instance, if you add 1 to an `int8` number 127, you get -128.  If the number gets promoted earlier, you get 128.  Natalia suggested removing int support for softsign.  I am not sure if more needs to be done besides changing the testing.
2. `rrelu` uses `uniform` underneath that does not seem to have its testing for randomness set up.  Therefore, I disabled some tests which it may or may not be correct to do.

### CODE REVIEW

File: torch/_refs/nn/functional/__init__.py [hunk 1]
```diff
@@ -146,6 +152,26 @@ def relu(a: TensorLikeType, inplace: bool = False) -> TensorLikeType:
...
+    """
+    Reference implementation of torch.nn.functional.hardsigmoid
+    """
+
+    if inplace:
+        raise NotImplementedError
+
+    le_neg3 = a <= -3
+    ge_3 = a >= 3
+    zeros = torch.zeros_like(a)
```
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:169]: I think instead of zeros_like and ones_like you should just be able to call where with a scalar (this may require a rebase?). Here's an example: 

https://github.com/pytorch/pytorch/blob/bc1fef96aff4aeffe7a7b39ef8b4ff467860df28/torch/_refs/nn/functional/__init__.py#L312

File: torch/_refs/nn/functional/__init__.py [hunk 2]
```diff
@@ -168,6 +194,21 @@ def leaky_relu(
...
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a",),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+)
+def logsigmoid(a: TensorLikeType) -> TensorLikeType:
+    """
+    Reference implementation of torch.nn.functional.logsigmoid
+    """
+
+    lt_0 = a - torch.log1p(torch.exp(a))
```
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:207]: Nice use of log1p

File: torch/_refs/nn/functional/__init__.py (hunk 1 above)
Comment (kevinstephano) [reply torch/_refs/nn/functional/__init__.py:169]: This appears to be a situation where `refs.where` works but `torch.where` does not.

```
    other = torch.true_divide(a, 6) + 0.5
    return refs.where(a <= -3, 0, refs.where(a >= 3, 1, other))
```

Comment (mruberry) [reply torch/_refs/nn/functional/__init__.py:169]: If that's the case we should just call refs.where explicitly, then. Although I thought scalar support was added to torch.where, too? 

```
torch.where(torch.tensor((True, False)), 0, 1)
: tensor([0, 1])
```

File: torch/_refs/nn/functional/__init__.py [hunk 3]
```diff
@@ -183,6 +224,27 @@ def mish(a: TensorLikeType, inplace: bool = False) -> TensorLikeType:
...
+    inplace: bool = False,
+) -> TensorLikeType:
+    """
+    Reference implementation of torch.nn.functional.rrelu
+    """
+
+    if inplace:
+        raise NotImplementedError
+
+    rhs = refs.uniform(a.shape, low=lower, high=upper, dtype=a.dtype, device=a.device)
```
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:244]: Doesn't this need to be multiplied by `a`? 

https://pytorch.org/docs/master/generated/torch.nn.RReLU.html#torch.nn.RReLU

File: torch/_refs/nn/functional/__init__.py [hunk 4]
```diff
@@ -183,6 +224,27 @@ def mish(a: TensorLikeType, inplace: bool = False) -> TensorLikeType:
...
+    return refs.where(refs.ge(a, 0), a, rhs)
```
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:245]: We should manually check that this has the same gradient and NaN propagation at zero as the original torch operator. (I think we had to write the relu ref to be `a < 0, 0, a` to get consistency)

File: torch/testing/_internal/common_methods_invocations.py [hunk 5]
```diff
@@ -14634,8 +14634,7 @@ def error_inputs_mean(op_info, device, **kwargs):
     UnaryUfuncInfo(
         'nn.functional.softsign',
         ref=lambda x: x / (np.abs(x) + 1),
-        dtypes=all_types_and_complex_and(torch.float16, torch.bfloat16),
-        dtypesIfCUDA=all_types_and_complex_and(torch.float16, torch.bfloat16, torch.bool),
+        dtypes=floating_and_complex_types_and(torch.float16, torch.bfloat16),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:14637]: Per offline discussion, let's revert this change

**Review by mruberry — COMMENTED** [review]
Overall looks great, @kevinstephano! 

Just a few small inline comments for your review

File: torch/_refs/nn/functional/__init__.py (hunk 3 above)
Comment (kevinstephano) [reply torch/_refs/nn/functional/__init__.py]: Ugh.  Math fixed.
